src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -289,61 +276,45 @@ def execute(self, args):
 
         # TODO: This fundamentally alters the way Knack.invocation works here. Cannot be customized
         # with an event. Would need to be customized via inheritance.
+
+        expanded_args = list(_explode_list_args(parsed_args))
+        cmd = parsed_args.func
+        self.cli_ctx.data['command'] = parsed_args.command
+        self.cli_ctx.data['safe_params'] = [(p.split('=', 1)[0] if p.startswith('--') else p[:2]) for p in args if
+                                            (p.startswith('-') and len(p) > 1)]
+
+        command_source = self.commands_loader.command_table[command].command_source
+
+        extension_version = None
+        extension_name = None
+        try:
+            if isinstance(command_source, ExtensionCommandSource):
+                extension_name = command_source.extension_name
+                extension_version = get_extension(command_source.extension_name).version
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+        telemetry.set_command_details(self.cli_ctx.data['command'], self.data['output'],
```

Comment:
I imagine the kind of telemetry we will be getting will now be different for commands using --ids. It seems like this way might be better as we send the whole command. 

cc/ @JasonRShaver

Reply:
Should be the same, as it only sets some fields. We just won't be setting the same thing multiple times.